src/kOS/Screen/TermWindow.cs

```diff
@@ -849,32 +849,37 @@ void TerminalGui(int windowId)
 
-            List<IScreenBufferLine> buffer = mostRecentScreen.Buffer; // just to keep the name shorter below:
+            // When loading a quicksave, it is possible for the teminal window to update even though
+            // mostRecentScreen is null.  If that's the case, just skip the screen update.
+            if (mostRecentScreen != null)
+            {
+                List<IScreenBufferLine> buffer = mostRecentScreen.Buffer; // just to keep the name shorter below:
 
-            // Sometimes the buffer is shorter than the terminal height if the resize JUST happened in the last Update():
-            int rowsToPaint = Math.Min(screen.RowCount, buffer.Count);
+                // Sometimes the buffer is shorter than the terminal height if the resize JUST happened in the last Update():
+                int rowsToPaint = Math.Min(screen.RowCount, buffer.Count);
 
-            for (int row = 0; row < rowsToPaint; row++)
-            {
-                IScreenBufferLine lineBuffer = buffer[row];
-                for (int column = 0; column < lineBuffer.Length; column++)
+                for (int row = 0; row < rowsToPaint; row++)
                 {
-                    char c = lineBuffer[column];
-                    if (c != 0 && c != 9 && c != 32 && c < fontArray.Length)
-                        ShowCharacterByAscii(c, column, row, reversingScreen,
-                                             charWidth, charHeight, screen.Brightness);
+                    IScreenBufferLine lineBuffer = buffer[row];
+                    for (int column = 0; column < lineBuffer.Length; column++)
+                    {
+                        char c = lineBuffer[column];
+                        if (c != 0 && c != 9 && c != 32 && c < fontArray.Length)
+                            ShowCharacterByAscii(c, column, row, reversingScreen,
+                                                 charWidth, charHeight, screen.Brightness);
+                    }
                 }
-            }
 
-            bool blinkOn = cursorBlinkTime < 0.5f &&
-                           screen.CursorRowShow < screen.RowCount &&
-                           IsPowered &&
-                           ShowCursor;
-            
-            if (blinkOn)
-            {
-                ShowCharacterByAscii((char)1, screen.CursorColumnShow, screen.CursorRowShow, reversingScreen,
-                                     charWidth, charHeight, screen.Brightness);
+                bool blinkOn = cursorBlinkTime < 0.5f &&
+                               screen.CursorRowShow < screen.RowCount &&
+                               IsPowered &&
+                               ShowCursor;
+
+                if (blinkOn)
+                {
+                    ShowCharacterByAscii((char)1, screen.CursorColumnShow, screen.CursorRowShow, reversingScreen,
+                                         charWidth, charHeight, screen.Brightness);
+                }
```

Comment:
I stared at this a while trying to figure out what you changed by deleting all those lines and adding these replacement lines that don't seem that different.. and then remembered that git's diff tool keeps showing mere indentation changes this way.  It seems to be capable of detecting mere indentations and showing them as such only when they last just a couple of lines.  When it's a whole block, it can't detect it.